src/axom/core/memory_management.hpp

```diff
@@ -200,23 +194,47 @@ inline void deallocate( T*& pointer ) noexcept
 template < typename T >
 inline T* reallocate( T* pointer, std::size_t n ) noexcept
 {
-  if ( n == 0 )
-  {
-    axom::deallocate( pointer );
-    return nullptr;
-  }
-
   const std::size_t numbytes = n * sizeof( T );
 
 #ifdef AXOM_USE_UMPIRE
 
+  // Workaround for bug in Umpire's handling on reallocate(0)
+  // Fixed in Umpire PR #292 (after v1.1.0)
+  if(n==0)
```

Comment:
Please update the documentation of this method as well and document the behavior when `n==0` in a `\note`.

Reply:
Will do.

Reply:
Thanks!

Reply:
Done.